src/main/java/org/terasology/thirst/ThirstAuthoritySystem.java

```diff
@@ -31,13 +31,23 @@
 import org.terasology.thirst.component.ThirstComponent;
 import org.terasology.thirst.event.DrinkConsumedEvent;
 
+/**
+ * This authority system handles drink consumption by various entities.
+ */
 @RegisterSystem(value = RegisterMode.AUTHORITY)
 public class ThirstAuthoritySystem extends BaseComponentSystem {
     @In
     private EntityManager entityManager;
     @In
     private Time time;
 
+    /**
+     * Initialize thirst attributes for a spawned player. Called when a player is spawned.
+     *
+     * @param event  the event corresponding to the spawning of the player
+     * @param player a reference to the player entity
+     * @param thirst a thirst component to assign to the player
```

Comment:
I think the player needs to have a `ThirstComponent` attached for this event handler. The _existing_ component is modified and saved. Probably something like "the player's thirst component to initialize".

**Note:** When referencing other classes, you can use `{@link org.terasology.thirst.component.ThirstComponent}` in the javadoc.